sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3264,18 +3264,13 @@ object GpuOverrides extends Logging {
           TypeSig.all))),
       (c, conf, p, r) => new TypedImperativeAggExprMeta[CollectSet](c, conf, p, r) {
 
-        private def isNestedArrayType(dt: DataType): Boolean = {
-          dt match {
-            case StructType(fields) =>
-              fields.exists { field =>
-                field.dataType match {
-                  case sdt: StructType => isNestedArrayType(sdt)
-                  case _: ArrayType => true
-                  case _ => false
-                }
-              }
-            case ArrayType(et, _) => et.isInstanceOf[ArrayType] || et.isInstanceOf[StructType]
-            case _ => false
+        override def tagAggForGpu(): Unit = {
```

Comment:
Why not just remove TypeSig.FLOAT and TypeSig.DOUBLE from the supported types?

```
 expr[CollectSet](
      "Collect a set of unique elements, not supported in reduction",
      // https://github.com/NVIDIA/spark-rapids/issues/8808 fall back in case there might be NaNs
      ExprChecks.fullAgg(
        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 +
            TypeSig.NULL + TypeSig.STRUCT + TypeSig.ARRAY - TypeSig.FLOAT - TypeSig.DOUBLE),
        TypeSig.ARRAY.nested(TypeSig.all),
        Seq(ParamCheck("input",
          (TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 +
              TypeSig.NULL +
              TypeSig.STRUCT +
              TypeSig.ARRAY - TypeSig.FLOAT - TypeSig.DOUBLE).nested(),
          TypeSig.all))),
...
```

This should update the docs and the make it more clear that we do not support these types.